src/Rules/Cast/PrintRule.php

```diff
@@ -43,16 +44,17 @@ static function (Type $type): bool {
 			}
 		);
 
+		$errors = [];
 		if (!$typeResult->getType() instanceof ErrorType
 			&& $typeResult->getType()->toString() instanceof ErrorType
 		) {
-			return [sprintf(
+			$errors[] = RuleErrorBuilder::message(sprintf(
 				'Parameter %s of print cannot be converted to string.',
 				$typeResult->getType()->describe(VerbosityLevel::value())
-			)];
+			))->line($node->getLine())->build();
```

Comment:
$node->expr->getLine()

Reply:
And the original structure should stay since only one error can be reported here.